File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    max_width = 0
    max_height = 0

    for size in sizes:
        if size[0] > size[1]:
            size[0], size[1] = size[1], size[0]

        max_width = max(size[0], max_width)
        max_height = max(size[1], max_height)
    
    return (max_width*max_height)

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    
    int max_width = 0;
    int max_height = 0;
    
    for (auto s: sizes) {
        int width = max(s[0], s[1]);
        int height = min(s[0], s[1]);
        
        if (max_width < width) max_width = width;
        if (max_height < height) max_height = height;
    }
    
    answer = max_width * max_height;
    
    return answer;
}